mailstream/mailstream.php

```diff
@@ -170,105 +175,141 @@ function mailstream_do_images($a, &$item, &$attachments) {
 	return $attachments;
 }
 
-function mailstream_sender($item) {
+function mailstream_sender(array $item) {
 	$r = q('SELECT * FROM `contact` WHERE `id` = %d', $item['contact-id']);
+
 	if (DBM::is_result($r)) {
-		$contact = $r[0];
-		if ($contact['name'] != $item['author-name']) {
-			return $contact['name'] . ' - ' . $item['author-name'];
+		if ($r[0]['name'] != $item['author-name']) {
+			return $r[0]['name'] . ' - ' . $item['author-name'];
 		}
 	}
+
 	return $item['author-name'];
 }
 
 function mailstream_decode_subject($subject) {
 	$html = BBCode::convert($subject);
+
 	if (!$html) {
 		return $subject;
 	}
+
 	$notags = strip_tags($html);
+
 	if (!$notags) {
 		return $subject;
 	}
+
 	$noentity = html_entity_decode($notags);
+
 	if (!$noentity) {
 		return $notags;
 	}
+
 	$nocodes = preg_replace_callback("/(&#[0-9]+;)/", function($m) { return mb_convert_encoding($m[1], "UTF-8", "HTML-ENTITIES"); }, $noentity);
+
 	if (!$nocodes) {
 		return $noentity;
 	}
+
 	$trimmed = trim($nocodes);
+
 	if (!$trimmed) {
 		return $nocodes;
 	}
+
 	return $trimmed;
 }
 
-function mailstream_subject($item) {
-	if ($item['title']) {
+function mailstream_subject(array $item) {
+	if (!empty($item['title'])) {
 		return mailstream_decode_subject($item['title']);
 	}
+
 	$parent = $item['thr-parent'];
 	// Don't look more than 100 levels deep for a subject, in case of loops
 	for ($i = 0; ($i < 100) && $parent; $i++) {
 		$r = q("SELECT `thr-parent`, `title` FROM `item` WHERE `uri` = '%s'", dbesc($parent));
+
 		if (!DBM::is_result($r)) {
 			break;
 		}
+
 		if ($r[0]['thr-parent'] === $parent) {
 			break;
 		}
+
 		if ($r[0]['title']) {
 			return L10n::t('Re:') . ' ' . mailstream_decode_subject($r[0]['title']);
 		}
+
 		$parent = $r[0]['thr-parent'];
 	}
+
 	$r = q("SELECT * FROM `contact` WHERE `id` = %d AND `uid` = %d",
 		intval($item['contact-id']), intval($item['uid']));
+
+	/// @TODO If above statement fails, this will cause E_NOTICE-s
+	/// @TODO Also maybe directly use $r[0] instead of variable copy?
```

Comment:
This should be a `dba::selectFirst()` call, feeding into `$contact` directly.

Reply:
Maybe leave for next PR?

Reply:
Sure.